### ISSUE TITLE
1Password SSH Agent not supporting key exchange with SHA1

### ISSUE THREAD

**Issue by tregota** [mscdex/ssh2#1186]
I have been trying to use 1Passwords SSH Agent in Windows but all I get is "Agent responded with failure".
It's being thrown from protocol.sign
I have read that 1Password doesn't support SHA1:
["A possible cause is the client/server is attempting key exchange with ssh-rsa (RSA with SHA1) which is not supported. 1Password currently only supports rsa-sha2-512 and rsa-sha2-256 for RSA keys."](https://1password.community/discussion/127277/just-tried-to-configure-1password-ssh-and-am-having-trouble-windows-11)

And I have noticed that the public key provided to the sign function in my test contain:
```
comment:'test'
Symbol(Decrypted Key):false
Symbol(Hash Algorithm):'sha1'
Symbol(Private key PEM):null
Symbol(Public key PEM):'-----BEGIN PUBLIC KEY-----[redacted]-----END PUBLIC KEY-----'
Symbol(Public key SSH):Buffer(535) [redacted]
type:'ssh-rsa'
```
this is for a 4096-bit RSA generated by 1Password.
Am I doing something wrong or is it the key parser or am I way off.

Thanks for a great module.
/Henrik

**Comment by tregota** [comment]
Nevermind, the signing actually works when running the latest 1password beta.
Still doesn't work for me sadly but one step closer.

**Comment by tregota** [comment]
Closed because I thought the 1Password beta fixed something, but it was only ssh-ed25519 that worked. RSA still doesn't play along. So I'm reopening hoping you have some input.

**Comment by mscdex** [comment]
Duplicate of https://github.com/mscdex/ssh2/issues/989.
